src/test/java/com/conveyal/gtfs/loader/FieldTests.java

```diff
@@ -30,24 +27,54 @@ public void dateFieldParseTest() {
                 "23000527", // Very distant future year
                 "790722" // Two digit year
         };
-
+        DateField dateField = new DateField("date", Requirement.REQUIRED);
         for (String badDate : badDates) {
-            try {
-                new DateField("date", Requirement.REQUIRED).validateAndConvert(badDate);
-                assertThat("Parsing bad dates should throw an exception and never reach this point.", false);
-            } catch (StorageException ex) {
-                assertThat("Error type should be date-related.",
-                        ex.errorType == NewGTFSErrorType.DATE_FORMAT || ex.errorType == NewGTFSErrorType.DATE_RANGE);
-                assertThat("Error's bad value should be the input value (the bad date).", ex.badValue.equals(badDate));
-            }
+            ValidateFieldResult<String> result = dateField.validateAndConvert(badDate);
+            assertThat("Parsing bad dates should result in an error.", result.errors.size() > 0);
+            NewGTFSError error = result.errors.iterator().next();
+            assertThat("Error type should be date-related.",
+                       error.errorType == NewGTFSErrorType.DATE_FORMAT || error.errorType == NewGTFSErrorType.DATE_RANGE);
+            assertThat("Error's bad value should be the input value (the bad date).", error.badValue.equals(badDate));
         }
 
         String[] goodDates = { "20160522", "20180717", "20221212", "20190505" };
 
         for (String goodDate : goodDates) {
-            String convertedValue = new DateField("date", Requirement.REQUIRED).validateAndConvert(goodDate);
-            assertThat("Returned value matches the well-formed date.", convertedValue.equals(goodDate));
+            ValidateFieldResult<String> result = dateField.validateAndConvert(goodDate);
+            assertThat("Returned value matches the well-formed date.", result.clean.equals(goodDate));
         }
 
     }
+
+    /**
+     * Make sure {@link Field#cleanString(ValidateFieldResult)} catches and removes illegal character sequences.
+     */
+    @Test
+    public void illegalCharacterParseTest() {
```

Comment:
Each of these tests should be refactored into[ parameritized tests](https://github.com/junit-team/junit4/wiki/parameterized-tests). This way is preferable because each test case will be ran regardless of whether a previous test case failed.

Reply:
Interesting and good to know about the parameterized test runner. It does seem like a more "correct" way to express running the same test N times with different input. It's unfortunate though that apparently the parameters are Objects instead of specific types, and I find the use of constructors to set fields on a class instance to be an awkward way to get the parameters into the test. But I suppose once you're used to seeing the pattern it's readable.

Reply:
I find this construction very awkward as well, and had simply copied the test for the DateField type directly above for this new illegal characters test. If both of these tests need revision, I suggest we pull a new ticket and handle this test issue separately.